sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncContainer.java

```diff
@@ -408,6 +416,29 @@ public String getId() {
         return queryItemsInternal(new SqlQuerySpec(query), new CosmosQueryRequestOptions(), classType);
     }
 
+    /**
+     * Initializes the container by warming up the caches and connections.
+     * <br>The execution of this method is expected to result in some RU charges to your account.
+     * The number of RU consumed by this request varies, depending on data consistency, size of the overall data in the container,
+     * item indexing, number of projections. For more information regarding RU considerations please visit
+     * <a href="https://docs.microsoft.com/en-us/azure/cosmos-db/request-units#request-unit-considerations">https://docs.microsoft.com/en-us/azure/cosmos-db/request-units#request-unit-considerations</a>.
+     * @return Mono of Void
+     */
+    @Beta(value = Beta.SinceVersion.V4_13_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+    public Mono<Void> openConnectionsAndInitCaches() {
+        SqlQuerySpec querySpec = new SqlQuerySpec();
+        querySpec.setQueryText("select * from c where c.id = @id");
```

Comment:
if the underlying container partitionKeyPath is "/id" then this query will act as a single partition query and we will only warming up the connections to a single parition.

Reply:
@mbhaskar thoughts ?

Reply:
Thats true. If "id" is partition key then `select * from c where c.id = @id` its prettymuch a single partition query

Reply:
I tested , it is not a single partition but yes also not covering all the partitions. Weird as it sound . Reverted to feed range query